pymodbus/client/asynchronous/async_io/__init__.py

```diff
@@ -880,6 +885,6 @@ def init_udp_client(proto_cls, loop, host, port, **kwargs):
     :return:
     """
     client = ReconnectingAsyncioModbusUdpClient(protocol_class=proto_cls,
-                                                loop=loop)
+                                                loop=loop, **kwargs)
```

Comment:
before this, remove from kwargs (i.e. {arg:argv[arg] for arg in kwargs if arg not in ["loop", "protocol_class"]}